Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -671,6 +671,18 @@ URL HitTestResult::absoluteLinkURL() const
     return url;
 }
 
+bool HitTestResult::hasLocalDataForLinkURL() const
+{
+    URL linkURL = absoluteLinkURL();
+    if (linkURL.isEmpty())
+        return false;
+
+    if (auto* page = m_innerURLElement->document().page())
```

Comment:
As we are starting to adopt smart pointers everywhere:
```suggestion
    if (RefPtr page = m_innerURLElement->document().page())
```